junit-platform-launcher/src/main/java/org/junit/platform/launcher/core/EngineDiscoveryOrchestrator.java

```diff
@@ -115,13 +119,14 @@ private LauncherDiscoveryResult discoverSafely(LauncherDiscoveryRequest request,
 	}
 
 	private TestDescriptor discoverEngineRoot(TestEngine testEngine, LauncherDiscoveryRequest request,
-			LauncherDiscoveryListener listener) {
+			LauncherDiscoveryListener listener, UniqueId parentId) {
 		UniqueId uniqueEngineId = UniqueId.forEngine(testEngine.getId());
+		UniqueId executionId = parentId == null ? uniqueEngineId : parentId.append(uniqueEngineId.getLastSegment()); // TODO: Do better
 		try {
-			listener.engineDiscoveryStarted(uniqueEngineId);
-			TestDescriptor engineRoot = testEngine.discover(request, uniqueEngineId);
+			listener.engineDiscoveryStarted(uniqueEngineId); // TODO: uniqueEngineId ok?
+			TestDescriptor engineRoot = testEngine.discover(request, executionId);
```

Comment:
Good question. I think we should pass in `executionId` because otherwise it would be inconsistent with what the engine passes to `selectorProcessed` and it's documented to be "the unique ID of the engine descriptor". I took a quick look and the internal implementation looked like they don't make any such assumptions.